src/main/java/uk/nhs/hee/tis/trainee/sync/facade/ProgrammeMembershipEnricherFacade.java

```diff
@@ -0,0 +1,625 @@
+      maxProgrammeCompletionDate = getNewMaximumProgrammeCompletionDate(maxProgrammeCompletionDate, thisProgrammeMembership);
+
+      // curricula
+      String curriculumId = getCurriculumId(thisProgrammeMembership);
+      if (curriculumId != null) {
+        Optional<Curriculum> optionalCurriculum = curriculumSyncService.findById(curriculumId);
+
+        if (optionalCurriculum.isPresent()) {
+          enrich(thisProgrammeMembership, optionalCurriculum.get());
+        } else {
+          doSync = false;
+          break;
+          // Cannot sync this record because all the related curriculum data is not available in local store.
+          // Note that we don't need to worry about programme data availability, since by definition the
+          // programmeMemberships will all have the same programmeId as programmeMembership, which has already been
+          // successfully enriched with locally-held programme data.
+        }
+      }
+      Set<Map<String, String>> thisCurricula = getCurricula(thisProgrammeMembership);
+      allCurricula.addAll(thisCurricula);
+    }
+
+    if (doSync) {
+      // final preparation and insertion of aggregate data
+
+      //TIS ID
+      List<String> sortedTisIds = new ArrayList<>(tisIds);
+      Collections.sort(sortedTisIds);
+      String allSortedTisIds = String.join(",", sortedTisIds);
+      aggregateProgrammeMembership.setTisId(allSortedTisIds);
+
+      // programmeCompletionDate
+      aggregateProgrammeMembership.getData().put(PROGRAMME_MEMBERSHIP_PROGRAMME_COMPLETION_DATE, String.valueOf(maxProgrammeCompletionDate));
+
+      // curricula
+      String allCurriculaJson = getCurriculaJson(allCurricula);
+      aggregateProgrammeMembership.getData().put(PROGRAMME_MEMBERSHIP_CURRICULA, allCurriculaJson);
+
+      // sync the complete aggregate programmeMembership record
+      tcsSyncService.syncRecord(aggregateProgrammeMembership);
+    }
+  }
+
+  /**
+   * Get the greater (most recent) date from a current maximum date and a programmeMembership's completion date.
+   *
+   * @param currentMaximumDate  The current maximum date
+   * @param programmeMembership The programme membership to retrieve the completion date from.
+   * @return The new maximum date.
+   *
+   */
+  private LocalDate getNewMaximumProgrammeCompletionDate(LocalDate currentMaximumDate,
+                                                         ProgrammeMembership programmeMembership) {
+    LocalDate newMaximumDate = currentMaximumDate;
+    String programmeCompletionDateString = getProgrammeCompletionDate(programmeMembership);
+    if (programmeCompletionDateString != null) {
+      LocalDate programmeCompletionDate = LocalDate.parse(programmeCompletionDateString);
+      if (currentMaximumDate == null) {
+        newMaximumDate = programmeCompletionDate;
+      } else {
+        if (programmeCompletionDate.isAfter(currentMaximumDate)) {
+          newMaximumDate = programmeCompletionDate;
+        }
+      }
```

Comment:
Looks like it could be condensed to 
```
if (currentMaximumDate == null || programmeCompletionDate.isAfter(currentMaximumDate)) {
  newMaximumDate = programmeCompletionDate;
}
```

Reply:
Yes, good call, fixed.